File: nvul_server/Models/CompilationRequest.cs
using nvul_compiler.Models.Configuration;
using System.ComponentModel.DataAnnotations;

namespace nvul_server.Models
{
	public class CompilationRequest
	{
		public string? NvulConfiguration { get; set; }
		public bool? ParsingResultRequired { get; set; }
		
		[Required]
		public string NvulCode { get; set; }

		public CompilationRequest(string nvulCode, string? nvulConfiguration)
		{
			this.NvulConfiguration = nvulConfiguration;
			this.NvulCode = nvulCode;
		}
	}
}

File: nvul_client/src/models/CompilationRequest.ts
export default class CompilationRequest {
    nvulConfiguration: string | null;
    parsingResultRequired: boolean | null;
    nvulCode: string;

    constructor(nvulCode: string, nvulConfiguration: string | null, parsingResultRequired: boolean | null) {
        this.nvulCode = nvulCode;
        this.nvulConfiguration = nvulConfiguration;
        this.parsingResultRequired = parsingResultRequired;
    }
}